datafusion/physical-expr/src/expressions/datetime.rs

```diff
@@ -220,6 +190,36 @@ impl PartialEq<dyn Any> for DateTimeIntervalExpr {
     }
 }
 
+/// create a DateIntervalExpr
+pub fn date_time_interval_expr(
```

Comment:
👍  this also follows the model of the other creation functions in physical expr.